### PR TITLE
AB#83231 - Users question is editable even if readonly mode

### PR DESCRIPTION
**Pull request by guisoares1**
# Description

To fix the problem, a behavior similar to text-widget was created. If there is any change in the readOnly attribute or in the survey, the field is disabled.

## Useful links

- [Link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/83231)

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [X] A grid was created with a reference data that contains the users field. After listing the records, click on details and observe the behavior.

## Screenshots

https://github.com/ReliefApplications/ems-frontend/assets/55603259/fb31075c-a234-407f-84e6-20f5c34dbadf

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

**Review by RenzoPrats — APPROVED** [review]
Working fine, looks good to me!

File: libs/shared/src/lib/survey/components/users.ts [hunk 1]
```diff
@@ -77,6 +83,16 @@ export const init = (
       instance.selectionChange.subscribe((value: string[]) => {
         question.value = value;
       });
+
+      (question.survey as SurveyModel).onValueChanged.add((survey: any) => {
+        if (survey.mode === 'display') instance.setReadOnly(true);
```
Comment (Matthis-M-ReliefApps) [inline libs/shared/src/lib/survey/components/users.ts]: Please always put bodies of if conditions between brackets, makes the code more consistent and clearer to read

File: libs/shared/src/lib/survey/components/users.ts [hunk 2]
```diff
@@ -66,7 +71,8 @@ export const init = (
       );
 
       const instance: UsersDropdownComponent = userDropdown.instance;
-
+      // set the visibility of the dropdown
```
Comment (Matthis-M-ReliefApps) [inline libs/shared/src/lib/survey/components/users.ts]: Should be a capital letter at the start (yes I'm over-picky)

**Review by Matthis-M-ReliefApps — CHANGES_REQUESTED** [review]
Just reviewed the code, didn't test it. Just requesting two very small style changes

**Comment by guisoares1** [comment]
@Matthis-M-ReliefApps  Requested changes made!

**Comment by AntoineRelief** [comment]
@RenzoPrats @guisoares1 
much simpler solution in last commit
to be honest, I haven't tested your code, and it would certainly have worked
I should have put in the ticket to use this kind of solution, which is much cleaner, as we directly use the built-in logic of the users-dropdown component & form wrapper directive
indeed, it will listen to changes on the form control, so you just need to take care of that in fact

**Pull request merged**